nbsite/cmd.py

```diff
@@ -1,7 +1,120 @@
 import os
+import glob
 
 from .util import copy_files
 
 def init(path):
     tmplate = os.path.join(os.path.dirname(__file__),'tmplate')
     copy_files(tmplate,path)
+
+
+git_hosts = {
+    # no trailing slash
+    'GitHub': 'https://raw.githubusercontent.com'
+}
+
+# TODO: really needs cleaning up! Currently just prototyping new behavior.
+def generate_rst(
+        project,
+        examples_path="./examples",
+        doc_path="./doc",
+#        title_formatter=None,
+        git_host='GitHub',
+        git_org=None,        
+        git_repo=None,
+        git_branch='master',
+        # legacy support
+        offset=0,        
+        overwrite=False):
+    """Auto-generates the rst files corresponding to the Notebooks in
+    examples_path
+
+    Takes title from filename, replacing underscores with spaces:
+    The_Title.ipynb -> The Title
+    01_The_Title.ipynb -> 01 The Title
+    01_Hyphen-Conscious_Title.ipynb -> 01 Hyphen-Conscious Title
+
+    and index.ipynb gets title from directory
+
+    Legacy options:
+      * offset: allows to skip leading n cells
```

Comment:
I don't think that's a legacy option; it should stay, even if we rarely use it. E.g. the PyViz tutorials all have a banner at the top of their first cell, which is useful for showing context when navigating between notebooks being run, but is not appropriate for building the web site.  Avoiding that option for most notebooks on holoviews.org, etc. is good, but there do seem to be valid uses for it.

Reply:
Plus I needed it for index.ipynb because without explicitly coding a title in .rst, the toctree wasn't correctly indented below a main section.